sources/Inputs/src/AutomatonConfigurationManager.cpp

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  AutomatonConfigurationManager.cpp
+ *
+ * @brief Automaton manager implementation file
+ *
+ */
+
+#include "AutomatonConfigurationManager.h"
+
+#include "Log.h"
+#include "Message.hpp"
+
+#include <boost/filesystem.hpp>
+#include <fstream>
+#include <iostream>
+#include <xml/sax/parser/ComposableDocumentHandler.h>
+#include <xml/sax/parser/Parser.h>
+#include <xml/sax/parser/ParserException.h>
+#include <xml/sax/parser/ParserFactory.h>
+
+namespace parser = xml::sax::parser;
+namespace file = boost::filesystem;
+
+namespace dfl {
+namespace inputs {
+
+namespace helper {
+
+template<class T>
+/**
+ * @brief Trait class for xsd filename by type
+ */
+struct XsdTrait {
```

Comment:
why "trait"?

Reply:
The template pattern used here is a known pattern named traits. It works as allocator_traits of stl. The pattern consists in adding members, type, etc based on a type for metaprogramming